uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/favorites/FavoritesUtilsTest.java

```diff
@@ -0,0 +1,91 @@
+        node1.setName("Cache Manager");
+        node1.setPrecedence(0.0);
+        node1.setMoveAllowed(true);
+        node1.setUnremovable(false);
+        node1.setDeleteAllowed(true);
+        node1.setImmutable(false);
+        node1.setEditAllowed(true);
+
+        UserLayoutChannelDescription node2 = new UserLayoutChannelDescription();
+        node2.setId("n28");
+        node2.setName("Permissions Administration");
+        node2.setPrecedence(0.0);
+        node2.setMoveAllowed(true);
+        node2.setUnremovable(false);
+        node2.setDeleteAllowed(true);
+        node2.setImmutable(false);
+        node2.setEditAllowed(true);
+
+        UserLayoutChannelDescription node3 = new UserLayoutChannelDescription();
+        node3.setId("n33");
+        node3.setName("User Administration");
+        node3.setPrecedence(0.0);
+        node3.setMoveAllowed(true);
+        node3.setUnremovable(false);
+        node3.setDeleteAllowed(true);
+        node3.setImmutable(false);
+        node3.setEditAllowed(true);
+
+        favArray.add(node1);
+        favArray.add(node2);
+        favArray.add(node3);
+
+        // Add the first node from favArray to favArrayWithDuplicates which creates the duplicate
+        // 'favorite' object in this Arraylist
+        favArrayWithDuplicate.add(favArray.get(0));
+
+        for (int i = 0; i < favArray.size(); i++) {
+            favArrayWithDuplicate.add(favArray.get(i));
+        }
+
+        List<IUserLayoutNodeDescription> uniqueFavorites =
+                FavoritesUtils.filterFavoritesToUnique(favArray);
+        List<IUserLayoutNodeDescription> favoritesReturnedFromUniqueFavoritesFilter =
+                FavoritesUtils.filterFavoritesToUnique(favArray);
+        List<IUserLayoutNodeDescription> emptyFavoritesListReturnedFromFavoritesFilter =
+                FavoritesUtils.filterFavoritesToUnique(emptyArray);
+
+        Assert.assertEquals(3, favArray.size());
+
+        // Confirms a list which is already unique will return the same
+        // list
+        Assert.assertEquals(3, favoritesReturnedFromUniqueFavoritesFilter.size());
+
+        // Confirms duplicate array contains 4 items, first two items
+        // being duplicates.
+        Assert.assertEquals(4, favArrayWithDuplicate.size());
+
+        // Confirms the new count of 3 after passing duplicate favorite
+        // list to filterFavoritesToUnique method
+        Assert.assertEquals(3, uniqueFavorites.size());
+
+        // Gets id of second element in duplicate array which before
+        // removing duplicate was "n18" and after it should be "n28"
+        String ObjectIdOfSecondItem = uniqueFavorites.get(1).getId();
+
+        // Confirms the first duplicate item was removed from the
+        // list
+        Assert.assertEquals("n28", ObjectIdOfSecondItem);
+
+        Assert.assertEquals(0, emptyArray.size());
+        Assert.assertEquals(0, emptyFavoritesListReturnedFromFavoritesFilter.size());
+    }
+}
```

Comment:
Excellent!!

Reply:
I'm wondering if this will works in the case we add channels in these nodes ? and having same channels in different nodes, that would be authorized, I think that your filter will remove "not duplicated" channels on different folders.

Could you add a such case in your tests ?

thanks !

Reply:
Always glad to add additional unit tests however I'm not sure I understand the scenario. By different folders do you mean folders other than the "favorites" folderType?

Reply:
@jgribonvald - I chatted offline with @bjagg a bit about this - if I understand your comment correctly, the favorites portlet should only show 1 entry per favorited portlet, no matter if it was favorited from different layouts / channels (thus @RamiroPuentesSCCCD should be able to base uniqueness solely on the portlet's `fname`).

Did I understand your comment correctly?

Reply:
@cbeach47: I think that some adopters are using the folder feature to organize favorites, in this case we can put as favorite a same portlet in several folders and so the filtering will permit to get only once the favorited portlet. Or I'm wrong ?
After In my mind using folders with favorites isn't a good idea, so this contribution is good for me.

Reply:
That's my understanding @jgribonvald - favorites portlet should ignore folders and just focus on the portlet (keyed by the `fname`).